### ISSUE TITLE
Example for enumerating classes/namespaces/functions

### ISSUE THREAD

**Issue by Unc3nZureD** [ChaiScript/ChaiScript#476]
Is it possible to do it from C++ side? If so, could you add an example?

Thank you!

**Comment by RobLoach** [comment]
What do you mean by enumerating classes/namespaces/functions? Are you trying to add classes, namespaces and functions to ChaiScript? Trying to add an enumeration?

**Comment by Unc3nZureD** [comment]
Sorry for poor description, let me try again :)

I meant that after loading a chaiscript file I'd like to find what namespaces/classes/functions the parser found, so that later on I can decide what functions I'd like to call.

For example if I implement a function in my chaiscript within a given namespace, then my CPP code can detect it and do something.
Note that I'm not interested in whether one particular function is implemented, but I'd like to make something dynamic, which compares what functions the Chai has & what features my CPP has and evaulate several chai code according to that.

**Comment by RobLoach** [comment]
I see these functions....

```
      m_engine.add(fun([this](){ m_engine.dump_system(); }), "dump_system");
      m_engine.add(fun([this](const Boxed_Value &t_bv){ m_engine.dump_object(t_bv); }), "dump_object");
      m_engine.add(fun([this](const Boxed_Value &t_bv, const std::string &t_type){ return m_engine.is_type(t_bv, t_type); }), "is_type");
      m_engine.add(fun([this](const Boxed_Value &t_bv){ return m_engine.type_name(t_bv); }), "type_name");
      m_engine.add(fun([this](const std::string &t_f){ return m_engine.function_exists(t_f); }), "function_exists");
      m_engine.add(fun([this](){ return m_engine.get_function_objects(); }), "get_functions");
      m_engine.add(fun([this](){ return m_engine.get_scripting_objects(); }), "get_objects");
```
